### PR TITLE
add Guest Service

### PR DESCRIPTION
**Pull request by Leana0**

### CODE REVIEW

File: src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        if(guest.getId()!= null && existsById(guest.getId())){
+            throw new EntityExistsException("Guest with id: " + guest.getId() + " already exists");
+        }
+        return guestRepository.save(guest);
+    }
+
+    @Transactional
+    public void update(Guest guest) {
+        notNull(guest, "Guest should not be null");
+        notNull(guest.getFirstName(), "Guest Name should not be null");
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java:50]: other fields?

File: src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+public class GuestService {
+
+    private final GuestRepository guestRepository;
+
+    private boolean existsById(Long id) {
+        return guestRepository.existsById(id);
+    }
+
+    public Guest findById(Long id) {
+        Guest guest = guestRepository.findById(id).orElse(null);
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java:26]: find by id and deleted is null

File: src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        if (guest == null) {
+            throw new EntityNotFoundException("Cannot find Guest with id: " + id);
+        } else {
+            return guest;
+        }
+    }
+
+    public List<Guest> findAll(final Pageable pageable) {return guestRepository.findAll(pageable).getContent();}
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java:34]: findAll and deleted is null

File: src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        guestRepository.save(guest);
+    }
+
+
+    public void deleteById(Long id) {
+        if (!existsById(id)) {
+            throw new EntityNotFoundException("Cannot find guest with id: " + id);
+        }
+        guestRepository.deleteById(id);
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java:62]: do not hard delete row, only update by setting deleted's value

File: src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    private boolean existsById(Long id) {
+        return guestRepository.existsById(id);
+    }
+
+    public Guest findById(Long id) {
+        Guest guest = guestRepository.findByIdAndDeletedIsNull(id).orElse(null);
+        if (guest == null) {
+            throw new EntityNotFoundException("Cannot find Guest with id: " + id);
+        } else {
```
Comment (nareknaltakyan1) [inline src/main/java/com/naltakyan/hotelmanagement/service/GuestService.java:31]: else is unnecessary, only write return statement

**Review by nareknaltakyan1 — APPROVED** [review]

**Pull request merged**